src/porepy/numerics/fem/p1.py

```diff
@@ -45,53 +45,63 @@ def ndof(self, g):
-            Values of the boundary conditions. The dictionary has at most the
-            following keys: 'dir' and 'neu', for Dirichlet and Neumann boundary
-            conditions, respectively.
+
+        We assume the following sub-dictionary to be present in the data dictionary:
+            parameter_dictionary, storing all parameters.
+                Stored in data[pp.PARAMETERS][self.keyword].
+
+        parameter_dictionary contains the entries:
+            second_order_tensor: (pp.SecondOrderTensor) Permeability defined cell-wise.
+            aperture: (np.ndarray) apertures of the cells for scaling of the face
+                normals.
+            bc: (pp.BoundaryConditionNode) node-wise boundary conditions.
+            bc_values: array (self.ndof) boundary condition values.
 
         Parameters
         ----------
-        g : grid, or a subclass, with geometry fields computed.
+        g: grid, or a subclass, with geometry fields computed.
         data: dictionary to store the data.
 
-        Return
+        Returns
         ------
-        matrix: sparse csr (g.num_nodes, g.num_nodes)
-            Matrix obtained from the discretization.
-        rhs: array (g.num_nodes)
-            Right-hand side which contains the boundary conditions and the scalar
-            source term.
-
+        matrix: sparse csr (g.num_nodes, g.num_nodes) Matrix obtained from the
+            discretization.
+        rhs: array (g.num_nodes) Right-hand side which contains the boundary conditions.
         """
         M, bc_weight = self.matrix(g, data, bc_weight=True)
         return M, self.rhs(g, data, bc_weight)
 
     # ------------------------------------------------------------------------------#
 
     def matrix(self, g, data, bc_weight=False):
-        """
-        Return the matrix for a discretization of a second order elliptic equation
-        using P1 method. See self.matrix_rhs for a detaild description.
+        """ Return the matrix for a discretization of a second order elliptic equation
+        using P1 method.
 
-        Additional parameter:
+        Parameters
+        ----------
+        g: grid, or a subclass, with geometry fields computed.
+        data: dictionary to store the data. See self.matrix_rhs for required contents.
+
+        Returns
+        ------
+        matrix: sparse csr (g.num_nodes, g.num_nodes)
+            Matrix obtained from the discretization.
+
+        Optional parameters:
         --------------------
-        bc_weight: to compute the infinity norm of the matrix and use it as a
-            weight to impose the boundary conditions. Default True.
+        is_tangential: bool, whether the lower-dimensional permeability tensor has been
+            rotated to the fracture plane. Defaults to False. Stored in data dictionary.
+        bc_weight: bool, whether to compute the infinity norm of the matrix and use it
+            as a weight to impose the boundary conditions. Default True.
 
```

Comment:
It defaults to False not True

Reply:
Well spotted. @alessiofumagalli and @keileg: note that this is for the matrix method, whereas it defaults to True in matrix_rhs. I suspect this could confuse others as well as me.